src/components/Slider/Slider.tsx

```diff
@@ -180,13 +180,14 @@ export class Slider extends BaseComponent<ISliderProps, ISliderState> implements
       currentValue = min + step * Math.round(currentSteps);
     }
 
+    if (onChange && currentValue !== this.state.value) {
```

Comment:
Also it would be nice to add a basic test that:

Renders a slider
Drags the thumb
Validates change event fires with the right value
Validates we can read the value prop from the component within the change callback